src/vehicleBooking/vehicleBooking.ts

```diff
@@ -0,0 +1,29 @@
+import logger from '../util/logger';
+import { VehicleBooking } from '../interfaces/VehicleBooking';
+import { vehicleBookingDb } from '../database/vehicleBookingDb';
+import { vehicleDb } from '../database/vehicleDb';
+import { VtBooking } from '../interfaces/VtBooking';
+
+export const vehicleBooking = {
+  async insert(vtBooking: VtBooking): Promise<void> {
+    logger.info('vehicleBooking insert starting');
+
+    const vehicle = await vehicleDb.get(vtBooking.vrm);
+
+    const booking = new VehicleBooking();
+    booking.SHORT_NAME = vtBooking.name;
+    booking.VEHICLE_CLASS = vehicle.VEHICLE_CLASS;
+    booking.NO_OF_AXLES = vehicle.NO_OF_AXLES;
+    booking.TIMESTAMP0 = new Date(vtBooking.bookingDate);
+    booking.VRM = vtBooking.vrm;
+    booking.FK_APPTYP_APPL_TYP = vtBooking.testCode;
+    booking.FK_LANTBD_DATE = new Date(vtBooking.testDate);
+    booking.FK_STATN_ID = vtBooking.pNumber;
+    booking.FK_VEH_SYST_NO = vehicle.SYSTEM_NUMBER;
+    booking.COUNTED_AXLES = vehicle.NO_OF_AXLES;
```

Comment:
```suggestion
    const booking = {
      ...(new VehicleBooking()),
      SHORT_NAME: vtBooking.name,
      VEHICLE_CLASS: vehicle.VEHICLE_CLASS,
      NO_OF_AXLES: vehicle.NO_OF_AXLES,
      TIMESTAMP0: new Date(vtBooking.bookingDate),
      VRM: vtBooking.vrm,
      FK_APPTYP_APPL_TYP: vtBooking.testCode,
      FK_LANTBD_DATE: new Date(vtBooking.testDate),
      FK_STATN_ID: vtBooking.pNumber,
      FK_VEH_SYST_NO: vehicle.SYSTEM_NUMBER,
      COUNTED_AXLES: vehicle.NO_OF_AXLES
    };
```

Reply:
I'm not a fan of this in C#. If one of the assignments has an issue, it's not clear in the stack trace which one was at fault. I'll see if the same issue occurs in JavaScript and amend if it doesn't.

Reply:
It's not an issue in Javascript.

Reply:
Thanks for looking into it mate 👍